graphql/resolvers/exerciseCrud.ts

```diff
@@ -102,6 +103,46 @@ export const deleteExercise = withUserContainer<
   })
 })
 
+export const removeExercise = withUserContainer<
+  Promise<Exercise>,
+  MutationRemoveExerciseArgs
+>(async (_parent: void, args: MutationRemoveExerciseArgs, ctx: Context) => {
+  const { req } = ctx
+  const { id } = args
+
+  const exercise = await prisma.exercise.findUnique({
+    where: {
+      id
+    }
+  })
+
+  if (!exercise) {
+    throw new Error('Exercise is not found')
+  }
+
+  const authorId = req.user?.id
+
+  if (!isAdmin(req) && exercise.authorId !== authorId) {
+    throw new Error('Not authorized to remove')
+  }
+
+  if (exercise.removedAt) {
+    throw new Error('Exercise is already removed')
+  }
+
+  return prisma.exercise.update({
+    where: { id },
+    data: {
+      removedAt: new Date().toISOString(),
```

Comment:
You don't need to convert to isostring, just passing a javascript date object is enough prisma will serialise it internally
e.g.
https://github.com/garageScript/c0d3-app/blob/cd8d8a595098ff376f734328fd08623d3c7b12a3/helpers/updateSubmission.ts#L108-L119